cdap-app-fabric/src/main/java/co/cask/cdap/app/DefaultAppConfigurer.java

```diff
@@ -275,8 +275,17 @@ public ApplicationSpecification createSpecification(@Nullable String application
                                                builtScheduleSpecs, workers, getPlugins());
   }
 
+  /**
+   * Adds the dataset and plugins at the application level configuration for a program
+   *
+   * @param configurer the program's configurer
+   */
   private void addDatasetsAndPlugins(AbstractConfigurer configurer) {
     addDatasets(configurer);
+    // Note: In case of DataPiplineApp the plugins are already registered in the application configuration.
```

Comment:
Shouldn't mention datapipline app in the core app-fabric code. Please make the comment generic.

Reply:
agreed. You can just say (in the javadoc section and not here) that a plugin or dataset registered by a program can be used by any other program in that application.

Reply:
Done.